src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/scarlet_monastery/instance_scarlet_monastery.cpp

```diff
@@ -40,15 +40,20 @@ void instance_scarlet_monastery::OnCreatureCreate(Creature* pCreature)
     switch (pCreature->GetEntry())
     {
         case NPC_MOGRAINE:
-            m_npcEntryGuidStore[pCreature->GetEntry()] = pCreature->GetObjectGuid();
             if (GetData(TYPE_ASHBRINGER_EVENT) == IN_PROGRESS)
                 DoOrSimulateScriptTextForThisInstance(SAY_ASHBRINGER_ENTRANCE, NPC_MOGRAINE);
-            break;
         case NPC_WHITEMANE:
         case NPC_VORREL:
             m_npcEntryGuidStore[pCreature->GetEntry()] = pCreature->GetObjectGuid();
             break;
     }
+    if (!pCreature->IsCritter()){
+        monasteryNPCs.push_back(pCreature);
+        if (GetData(TYPE_ASHBRINGER_EVENT) == IN_PROGRESS)
```

Comment:
This might seem redundant, but not all creatures spawn when the user enters, some of them spawn way later.
If I *only* do this OnPlayerEnter, all NPCs in the chapel will remain hostile. If I *only* do this OnCreatureCreate: the first creatures will be created before the ASHBRINGER_EVENT is set to `IN_PROGRESS` so they also wouldn't get the new faction.